keanu-project/src/main/java/io/improbable/keanu/vertices/Covariance.java

```diff
@@ -0,0 +1,86 @@
+package io.improbable.keanu.vertices;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.nd4j.linalg.api.blas.BlasException;
+
+import com.google.common.collect.ImmutableList;
+
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.dbl.ScalarDoubleTensor;
+
+public class Covariance {
```

Comment:
I needed a new class so that I can get a smaller covariance matrix for a subset of the dimensions. This is necessary as the MCMC algorithms can select a subset of vertices as a proposal.